pages/docs/tutorials/getting-started/slackbot-websocket-tutorial.md

```diff
@@ -0,0 +1,340 @@
+participant Slack Server (Socket Mode)
+participant User
+
+
+Heart-Counter-Slack-Bot ->>Slack Server (Socket Mode): Connect
+Note right of Heart-Counter-Slack-Bot: Establish WebSocket connection
+
+
+Slack Server (Socket Mode)->> Heart-Counter-Slack-Bot : Handshake
+Note left of Slack Server (Socket Mode): Establish Socket Mode connection
+
+Slack Server (Socket Mode)->> Heart-Counter-Slack-Bot : Sends "hello" event
+Note left of Slack Server (Socket Mode): Confirms successful connection
+
+
+
+
+
+User-->>Slack Server (Socket Mode): Adds emoji to a message.
+
+
+Slack Server (Socket Mode)->>Heart-Counter-Slack-Bot: Sending "reaction_added" payload
+
+
+Note left of Heart-Counter-Slack-Bot: Event received
+``` 
+
+## Background context
+[WebSocket](https://en.wikipedia.org/wiki/WebSocket) is a communication protocol that enables simultaneous bidirectional data exchange between a client and a server over a single, long-lived connection. Unlike HTTP, which relies on the request-response model, WebSocket is ideal for scenarios where real-time, interactive and low-latency communication is necessary.
+
+
+```mermaid
+sequenceDiagram
+Client -->> Server: Initiate WebSocket Connection
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+Server -->> Client: WebSocket Response
+```
+
+In Slack, WebSocket is employed as part of its [Socket Mode](https://api.slack.com/apis/connections/socket) feature to facilitate real-time notifications between Slack's servers and third-party applications or bots. The [Slack Event API](https://api.slack.com/apis/connections/events-api) is a tool that lets you receive real-time notifications of specific events in a Slack workspace such as messages, reactions, and user presence changes.
+
+
+## Define AsyncAPI Version, API Information, and Server
+
+You start your AsyncAPI document by specifying the AsyncAPI version and essential information about your Slack application's API which includes details such as the `title`, `version` and `description`.
+
+The `ws` server section allows you to define the protocol and specify information about the URLs your application will use, such as  `host`, `protocol` and `description`.
+
+<Remember>
+The WebSocket URL is generated  by invoking the <a href="https://api.slack.com/methods/apps.connections.open">apps.connections.open</a> method from Slack’s API. You use the authentication tokens obtained during the configuration of  <code>Heart-Counter-Slack-Bot</code> to generate this URL.
+</Remember>
+
+```
+asyncapi: '3.0.0'
+info:
+  title: Create an AsyncAPI Document for a SlackBot with WebSockets
+  version: '1.0.0'
+  description:  |
+    The Heart-Counter-Slack-Bot manages popular messages in a Slack workspace by monitoring message reaction data using Slack's Event API.
+    
+servers:
+  ws:
+    host: wss://wss-primary.slack.com/
+    protocol: wss
+    description: Websocket URL generated to communicate with Slack
```

Comment:
this should be the description of your server. something like `Slack Socket Mode server, used to receive events from a slack workspace.`